src/components/organims/welcomeByOrg.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {Redirect, navigate} from '@reach/router';
+import AppMinimal from '../layout/AppMinimal';
+import Seo from '../molecules/Seo';
+import Button from '../atoms/Button';
+import styles from './welcome.module.css';
+import gql from 'graphql-tag';
+
+const getOrgInfo = gql`
+  query($slug: String!) {
+    organizationWithSlug(slug: $slug) {
+      slug,
+      name,
+      pictureUrl,
+    }
+  }
+`;
+
+export class WelcomeByOrg extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = {
+      title: 'Your organization name',
+      description: 'Your organization description',
+      details: 'Your organization details',
+      email: 'name.org',
+      pictureUrl: '',
+      hasDownloaded: false,
+    }
+  }
+  componentWillMount(){
```

Comment:
The difficulty I saw here is that `Pages` has some logic with Gatsby, so that when you go to the page `welcome` you are re-routed to the `welcomeByOrg` component & it's only this last which can check  it's params because of the magic of Gatsby [set here](https://github.com/getstation/website-v3/pull/38/files#diff-fda05457e393bada716f508859bfc604R26) + [router](https://github.com/getstation/website-v3/pull/38/files#diff-efa89f1bed3ade1e9818de329a3bbb5cR27)

Reply:
>  so that when you go to the page welcome you are re-routed to the welcomeByOrg component
I don't understand the meaning of _rerouted to a component_ here.

Reply:
Anyway, you really need to find a way to remove all the logic (calling API, downloading..) outside. Do you see a way?